lib/matplotlib/axes/_axes.py

```diff
@@ -6724,7 +6741,8 @@ def hist(self, x, bins=None, range=None, density=None, weights=None,
                     height = m
                 patch = _barfunc(bins[:-1]+boffset, height, width,
                                  align='center', log=log,
-                                 color=c, **{bottom_kwarg: bottom})
+                                 color=c, edgecolor=ec,
```

Comment:
This gets into the `color`, `facecolor`, `edgecolor` confusion.    Probably should be clear in the doc of `color` what the color means, and it'd be nice if it was consistent w/ the other instances of this.  (Yes this ambiguity predated this PR due to the fact that patch kwargs are passed through to patch).

Reply:
I'll look into linewidth and the color descriptions.

It's even a bit worse since the meaning of `color` depends on the `histtype`. For `histtype='bar'` and `histtype='stepfilled'` it's the facecolor. For `histtype='step'` it's the edgecolor and the `edgecolor` is ignored. Even though, this behavior is confusing, I intentionally did not touch it in this PR to keep backward compatibility.

Reply:
Fair enough but the color doc string could be made more specific.  When we didn’t have an edgecolor kwarg we could assume some expertise in the part of the user who were able to drill down to the patch kwargs.  But with that kwarg it’s going to lead to confusion.

Reply:
Reworked the documentatio of the color parameter.